oligo_designer_toolsuite/oligo_specificity_filter/_filter_bowtie.py

```diff
@@ -0,0 +1,337 @@
+        :rtype: pandas.DataFrame
+        """
+
+        def _run_bowtie_batch(batch_id):
+            """Run Bowtie alignment tool for all probes of one batch
+            :param batch_id: Batch ID.
+            :type batch_id: int
+            """
+
+            for subbatch_id in range(self.number_subbatches):
+                file_probe_fasta_batch = os.path.join(
+                    self.dir_annotations,
+                    "probes_sequence_batch{}_{}.fna".format(batch_id, subbatch_id),
+                )
+                file_bowtie_batch = os.path.join(
+                    self.dir_bowtie,
+                    "bowtie_batch{}_{}.txt".format(batch_id, subbatch_id),
+                )
+
+                if self.mismatch_region is not None:
+                    command = (
+                        "bowtie transcriptome -f -a -n "
+                        + str(self.min_mismatches - 1)
+                        + "-l"
+                        + str(self.mismatch_region)
+                        + " "
+                        + file_probe_fasta_batch
+                        + " "
+                        + file_bowtie_batch
+                    )
+                else:
+                    command = (
+                        "bowtie transcriptome -f -a -v "
+                        + str(self.min_mismatches - 1)
+                        + " "
+                        + file_probe_fasta_batch
+                        + " "
+                        + file_bowtie_batch
+                    )
+
+                process = Popen(command, shell=True, cwd=self.dir_bowtie).wait()
+
+        index_exists = False
+        # Check if bowtie index exists
+        for file in os.listdir(self.dir_bowtie):
+            if re.search("^transcriptome*", file):
+                index_exists = True
+                break
+
+        # Create bowtie index if none exists
+        if not index_exists:
+            command1 = (
+                "bowtie-build " + self.file_transcriptome_fasta + " transcriptome"
+            )
+            process = Popen(command1, shell=True, cwd=self.dir_bowtie).wait()
+
+        start_time = time.perf_counter()
+
+        with parallel_backend("loky"):
+            Parallel()(
+                delayed(_run_bowtie_batch)(batch_id) for batch_id in range(self.n_jobs)
+            )
+
+        finish_time = time.perf_counter()
+        self.logging.info(
+            f"Bowtie alignmnet search finished in {finish_time-start_time} seconds"
+        )
+
+    def filter_probes_by_bowtie_results(self):
+        """Use bowtie results to filter probes based on number of mismatches and mismatch position
+
+        :return: Text file for each gene containing the filtered probes for that gene. Additionally, a text file containing a list of genes with insufficient number of probes is returned.
+        :rtype: Txt files"""
+
+        def _process_bowtie_results(batch_id):
```

Comment:
this function looks quite familiar to the process_blast_results function. Can we merge this?